### DISCUSSION TITLE
MKL16Z64VLH4 support

### DISCUSSION THREAD

**Discussion by wisess** [blackmagic-debug/blackmagic#1343] [Ideas]
Hello! What about MKL16Z64VLH4 support? A lot of people have flysky i6 with this processor. Maybe somebody's added it in BMP?

**Comment by dragonmux** [comment]
Support for that part does not yet exist, but would be a welcome addition

**Reply by wisess** [reply]
What's needed for it?

**Reply by dragonmux** [reply]
Most likely, looking at [the NXP website](https://www.nxp.com/part/MKL16Z64VLH4#/), addition of a `mkl16.c` using the [new target guide](https://black-magic.org/hacking/target-drivers.html) to write support atop the Cortex-M support, and addition of suitable probe hooks to [cortexm.c](https://github.com/blackmagic-debug/blackmagic/blob/eca63c1fb332f4dd16ab0f67f536c5f00e3b3f9f/src/target/cortexm.c#L638-L731)

https://github.com/blackmagic-debug/blackmagic/blob/eca63c1fb332f4dd16ab0f67f536c5f00e3b3f9f/src/target/cortexm.c#L638-L731

We would embark on this ourself (once we got i.MXRT support done) but we don't presently have parts and NXP's parts are notorious for their errata so testing the support is imperative.

**Reply by wisess** [reply]
I will try to figure out it. But when you will have possibility let try to add it. Thank you!
